File: module_java/src/main/java/org/gmfbilu/superapp/module_java/singleton/Singleton1.java
package org.gmfbilu.superapp.module_java.singleton;

/**
 * 懒汉式非线程安全
 */
public class Singleton1 {

    private static Singleton1 sSingleton1;

    private Singleton1() {
    }

    public static Singleton1 getSingleton1() {
        if (sSingleton1==null){
            sSingleton1 = new Singleton1();
        }
        return sSingleton1;
    }
}

File: module_kotlin/src/main/java/org/gmfbilu/superapp/module_kotlin/kotlin/慕课网kotlin教程/设计模式/Singleton1.kt
package org.gmfbilu.superapp.module_kotlin.kotlin.慕课网kotlin教程.设计模式

/**
 * 懒汉式非线程安全
 */
class Singleton1 private constructor() {

    companion object {
        //LazyThreadSafetyMode.NONE初始化非线程安全
        val instance by lazy(LazyThreadSafetyMode.NONE) {
            Singleton1()
        }
    }
}